### PR TITLE
implement SearchBookItem component

### PR DESCRIPTION
**Pull request by tranchinh123**

### CODE REVIEW

File: bookshelter-app/src/components/SearchInput/index.tsx [hunk 1]
```diff
@@ -1,15 +1,65 @@
...
+  setResults: (results: Book[]) => void;
+  onOpen: () => void;
+  onclose: () => void;
+}
+
+const SearchInput = ({ setResults, onOpen, onclose }: SearchInputProps) => {
+  const [searchTerm, setSearchTerm] = useState('');
+  const navigate = useNavigate();
+
+  useEffect(() => {
```
Comment (huypham-agilityio) [inline bookshelter-app/src/components/SearchInput/index.tsx]: don't handle logic in component, let's move it to parent component

File: bookshelter-app/src/components/SearchInput/index.tsx [hunk 2]
```diff
@@ -1,15 +1,65 @@
...
+    const fetchBookList = async (): Promise<void> => {
+      const books = await get<Book[]>(API.BOOKS_ENDPOINT);
```
Comment (huypham-agilityio) [inline bookshelter-app/src/components/SearchInput/index.tsx]: use https://github.com/mockapi-io/docs/wiki/Code-examples#filtering to query data

File: bookshelter-app/src/pages/Home/index.tsx [hunk 3]
```diff
@@ -4,22 +4,43 @@ import { get } from '../../services/api';
...
+import { useParams, useNavigate } from 'react-router-dom';
+
 const HomePage = () => {
   const [books, setBooks] = useState<Book[]>([]);
+  const { slug, name } = useParams<{ slug: string; name: string }>();
+  const navigate = useNavigate();
 
   useEffect(() => {
     const fetchBookList = async (): Promise<void> => {
       const listBooks = await get<Book[]>(API.BOOKS_ENDPOINT);
```
Comment (huypham-agilityio) [inline bookshelter-app/src/pages/Home/index.tsx]: use https://github.com/mockapi-io/docs/wiki/Quick-start-guide#filtering to get book by category

File: bookshelter-app/src/components/SearchInput/index.tsx (hunk 1 above)
Comment (tranchinh123) [reply bookshelter-app/src/components/SearchInput/index.tsx]: Fixed!

File: bookshelter-app/src/components/SearchInput/index.tsx (hunk 2 above)
Comment (tranchinh123) [reply bookshelter-app/src/components/SearchInput/index.tsx]: Fixed

File: bookshelter-app/src/pages/Home/index.tsx (hunk 3 above)
Comment (tranchinh123) [reply bookshelter-app/src/pages/Home/index.tsx]: fixed!

**Review by huypham-agilityio — APPROVED** [review]

**Pull request merged**